### DISCUSSION TITLE
Parametrised tests: Run a single invocation via maven

### DISCUSSION THREAD

**Discussion by AlexShmydov** [junit-team/junit-framework#3794] [Q&A] [answered]
Hi all!

### Question:
How to run a single invocation of a parameterised test via maven? 

i can re-run a single invocation of a parameterised test in IDEA in accord with https://github.com/junit-team/junit5/issues/1025, but i can not do that via maven, for example with uniqueId.

**Comment by AlexShmydov** [comment]
**My question relates to Junit 5 (version 5.10.1)**

As i see, Junit4/Juni4.7+ provide possibility to run single invocation test in accord with https://maven.apache.org/surefire/maven-surefire-plugin/examples/single-test.html

`If using the JUnit annotation @Parameters(name="{index}: fib({0})={1}") and selecting the index e.g. 5 in pattern, the non-regex method pattern would become #testMethod[5:*].`

but it does not work on Junit 5

**Comment by marcphilipp** [comment, marked as answer]
JUnit Platform does provide the means to implement this via `IterationSelector`. Parsing expressions like `#testMethod[5:*]` is a feature of Maven Surefire that's probably not implemented for JUnit 5, yet. I'd recommend raising an issue with the Maven Surefire project.

**Reply by AlexShmydov** [reply]
Thanks for answer. I found PR with this problem: https://github.com/apache/maven-surefire/pull/476
